scripts/run-pyright.py

```diff
@@ -137,19 +150,35 @@ def load_path_file(path: str) -> Sequence[str]:
 
 
 def normalize_args(args: argparse.Namespace) -> Args:
-    if args.all and (args.env or args.paths):
-        raise Exception("Cannot target specific environments or paths simultaneously with --all.")
+    if args.all and (args.diff or args.env or args.paths):
+        raise Exception(
+            "Cannot target specific environments, paths, or diff simultaneously with --all."
+        )
+    elif args.diff and (args.env or args.paths):
+        raise Exception("Cannot target specific environments or paths, simultaneously with --diff.")
     elif len(args.paths) >= 1 and len(args.env) >= 1:
         raise Exception("Cannot pass both paths and environments.")
-    use_all = args.all or not args.env and not args.paths
+    use_all = args.all or not (args.diff or args.env or args.paths)
     if args.env or use_all:
         envs = os.listdir(PYRIGHT_ENV_ROOT) if use_all else args.env or ["master"]
+        paths = []
         for env in envs:
             if not os.path.exists(get_env_path(env)):
                 raise Exception(f"Environment {env} not found in {PYRIGHT_ENV_ROOT}.")
+    elif args.diff:
+        paths = (
+            subprocess.check_output(["git", "diff", "--name-only", "HEAD"])
```

Comment:
since pyright is generally fast i think it would be useful to cover more ground here - like diff against the branch base or master or something. Would be nice for it to catch developing PR issues outside just before commit.

Reply:
either way i think it should output what it diffed against and what files it got as a result clearly in the output

Reply:
good ideas